tests/test_data.py

```diff
@@ -146,11 +148,42 @@ def test_valid_transitions(
         transitions: types.Transitions,
         transitions_rew: types.TransitionsWithRew,
         length: int,
+        n_checks: int = 20,
     ) -> None:
-        """Checks trajectories can be created for a variety of lengths and spaces."""
-        assert len(transitions_min) == length
-        assert len(transitions) == length
-        assert len(transitions_rew) == length
+        """Checks initialization, indexing, and slicing sanity."""
+        for trans in [transitions_min, transitions, transitions_rew]:
+            assert len(trans) == length
+
+            for _ in range(n_checks):
+                # Indexing checks, which require at least one element.
+                if length != 0:
+                    index = np.random.randint(length)
+                    assert isinstance(trans[index], dict)
+                    self._check_transitions_get_item(trans, index)
+
+                # Slicing checks.
+                start = np.random.randint(-2, length)
+                stop = np.random.randint(0, length + 2)
+                step = np.random.randint(-2, 4)
+                if step == 0:  # Illegal. Quick fix that biases tests to ordinary step.
+                    step = 1
+                s = slice(start, stop, step)
+                assert type(trans[s]) is type(trans)
+                self._check_transitions_get_item(trans, s)
+
+    def _check_transitions_get_item(self, trans, key):
```

Comment:
Why is this an instance method? It does not depend on `self` as far as I can see, so it could be a module level function instead?

Reply:
That makes sense, I think that I just wanted group this close to the function that used it. Moving it to module level because it's more google-styley